src/SpreadingDye.jl

```diff
@@ -21,7 +21,7 @@ function center_cell(m::AbstractMatrix{<:Number})
 end
 
 # is a point on the domain?
-within_edges(dom::AbstractMatrix, point::Tuple{Int, Int}) = min(point...) > 0 && first(point) <= size(dom, 1) && last(point) <= size(dom, 2)
+within_edges(dom::AbstractMatrix, point::Tuple{Int, Int}) = Base.checkbounds(Bool, dom, CartesianIndex(point))
```

Comment:
this is a nice change. In principle it is unrelated, so should go in a separate PR, but this does not cause any confusion